src/lib/dhcp/option6_auth.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2011-2016 Internet Systems Consortium, Inc. ("ISC")
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef OPTION6_AUTH_H
+#define OPTION6_AUTH_H
+#endif
+
+#include <dhcp/option.h>
+#include <boost/shared_ptr.hpp>
+
+#include <vector> 
+
+namespace isc {
+namespace dhcp {
+
+class Option6Auth;
+
+/// A pointer to the @c isc::dhcp::Option6Auth object.
+typedef boost::shared_ptr<Option6Auth> Option6AuthPtr;
+
+class Option6Auth: public Option {
+
+public:
+    
+    static const uint8_t OPTION6_AUTH_MIN_LEN  = 11;
+    static const uint8_t OPTION6_HASH_MSG_LEN  = 16;
+    static const uint8_t OPTION6_HDR = 4; 
+    /// @brief Constructor, used for auth options while transmitting
+    ///
+    /// @param proto protocol type
+    /// @param algo algorithm type
+    /// @param method remote detection method
+    /// @param rdm replay detection value
+    /// @param info authentication info.
+    Option6Auth(const uint8_t proto, const uint8_t algo, const uint8_t method,
+                const uint64_t rdm, const std::vector<uint8_t>& info);
+
+    /// @brief Copies this option and returns a pointer to the copy.
+    virtual OptionPtr clone() const;
+
+    /// Writes option in wire-format to buf, returns pointer to first unused
+    /// byte after stored option.
+    ///
+    /// @param buf buffer (option will be stored here)
+    void pack(isc::util::OutputBuffer& buf) const;
+
+    /// Writes option in wire-format to buf, for computing hash  
+    /// auth info filled with 0 for a length of 128 bits
+    /// returns with pointer to first unused
+    /// byte after stored option.
+    ///
+    /// @param buf buffer (option will be stored here)
+    void packHashInput(isc::util::OutputBuffer& buf) const;
```

Comment:
I think it will be necessary to return a pointer here to the place where the hash should be stored. But if I'm right, we can tweak this later.